src/main/drivers/accgyro/gyro_sync.c

```diff
@@ -49,21 +49,40 @@ bool gyroSyncCheckUpdate(gyroDev_t *gyro)
 
 uint32_t gyroSetSampleRate(gyroDev_t *gyro, uint8_t lpf, uint8_t gyroSyncDenominator, bool gyro_use_32khz)
 {
-    bool lpfNoneOr256 = (lpf == GYRO_LPF_256HZ || lpf == GYRO_LPF_NONE);
-    if (!lpfNoneOr256) {
-        gyroSyncDenominator = 1; // Always full Sampling
-    }
-    gyro->mpuDividerDrops = gyroSyncDenominator - 1;
-    gyro->gyroRateKHz = lpfNoneOr256 ? GYRO_RATE_8_kHz : GYRO_RATE_1_kHz;
-    //20649 is a weird gyro
-    if (gyro->mpuDetectionResult.sensor == ICM_20649_SPI) {
-        gyro->gyroRateKHz = lpfNoneOr256 ? GYRO_RATE_9_kHz : GYRO_RATE_1100_Hz;
-    } else if (gyro->mpuDetectionResult.sensor == BMI_160_SPI && lpfNoneOr256) { 
-        //brainFPV is also a weird gyro
-        gyro->gyroRateKHz = GYRO_RATE_3200_Hz;
-    } else if (gyro_use_32khz) {
-        //use full 32k
-        gyro->gyroRateKHz = GYRO_RATE_32_kHz;
+    float gyroSamplePeriod;
+
+    if (lpf != GYRO_HARDWARE_LPF_1KHZ_SAMPLE) {
+        if (gyro_use_32khz) {
+            gyro->gyroRateKHz = GYRO_RATE_32_kHz;
+            gyroSamplePeriod = 31.25f;
+        } else {
+            switch (gyro->mpuDetectionResult.sensor) {
+            case BMI_160_SPI:
+                gyro->gyroRateKHz = GYRO_RATE_3200_Hz;
+                gyroSamplePeriod = 312.0f;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'gyroSamplePeriod' is assigned a value that is never used.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5552604)